core/src/test/scala/unit/kafka/log/LogValidatorTest.scala

```diff
@@ -1398,6 +1437,73 @@ class LogValidatorTest {
     assertEquals(6, e.recordErrors.size)
   }
 
+  @Test
+  def testRecordWithPastTimestampIsRejected(): Unit = {
+    val timestampBeforeMaxConfig = 24 * 60 * 60 * 1000L //24 hrs
+    val timestampAfterMaxConfig = 1 * 60 * 60 * 1000L //1 hr
+    val now = System.currentTimeMillis()
+    val fiveMinutesBeforeThreshold = now - timestampBeforeMaxConfig - (5 * 60 * 1000L)
+    val records = createRecords(magicValue = RecordBatch.MAGIC_VALUE_V2, timestamp = fiveMinutesBeforeThreshold,
+      codec = CompressionType.GZIP)
+    val e = assertThrows(classOf[RecordValidationException],
+      () => new LogValidator(
+        records,
+        topicPartition,
+        time,
+        CompressionType.GZIP,
+        CompressionType.GZIP,
+        false,
+        RecordBatch.MAGIC_VALUE_V1,
+        TimestampType.CREATE_TIME,
+        timestampBeforeMaxConfig,
+        timestampAfterMaxConfig,
+        RecordBatch.NO_PARTITION_LEADER_EPOCH,
+        AppendOrigin.CLIENT,
+        MetadataVersion.latest
+      ).validateMessagesAndAssignOffsets(
+        PrimitiveRef.ofLong(0L), metricsRecorder, RequestLocal.withThreadConfinedCaching.bufferSupplier
+      )
+    )
+
+    assertTrue(e.invalidException.isInstanceOf[InvalidTimestampException])
+    assertFalse(e.recordErrors.isEmpty)
+    assertEquals(e.recordErrors.size, 3)
+  }
+
+
+  @Test
+  def testRecordWithFutureTimestampIsRejected(): Unit = {
+    val timestampBeforeMaxConfig = 24 * 60 * 60 * 1000L //24 hrs
+    val timestampAfterMaxConfig = 1 * 60 * 60 * 1000L //1 hr
+    val now = System.currentTimeMillis()
```

Comment:
In the tests we usually use MockTime. Is it possible to use that here?

Reply:
that will make the test a little bit involved. The reason is we are using a producer from the common test until `TestUtils.createProducer` which spins up a test producer using the System's current timestamp. We can potentially refactor that code so that it takes a mock time. But without that the timestamp validation will use the system's current timestamp hence the tests follow the same pattern as well.